### DISCUSSION TITLE
Adding TRUNCATE option to the api...

### DISCUSSION THREAD

**Discussion by jeffmikels** [pocketbase/pocketbase#560] [Q&A]
I'm adding a collection TRUNCATE option to the api.

I'm planning to use the following endpoint

- PATCH /api/collections/[name]/truncate (with an empty body)

Does anyone see a problem with that?

**Comment by ganigeorgiev** [comment]
Is this for a  custom app endpoint or you are planning to add it to the `pocketbase.apis` subpackage? If the latter, please keep in mind that PRs for new features are temporary "frozen" due to #376 since there are a lot of changes to the internals (especially to the integration tests) and it is better to wait.

If this endpoint is intended only for your app, you may want to add some unique prefix/suffix to your `/truncate` path to avoid collisions with future endpoints (I'll consider documenting it somewhere in the future).

_btw. If the truncate is expected to act as bulk delete for the collection records, note that there are plans also to add bulk create/update/delete in the future as part of #48 (you can also check this discussion https://github.com/pocketbase/pocketbase/discussions/545#discussioncomment-3677180)_

**Reply by jeffmikels** [reply]
I added it to my fork before I saw that PRs are frozen.

**Comment by synapse** [comment]
I am also looking for something like this. In my case I have a logs/history table which I need to clear all at once instead of doing it one-by-one which would mean I have just 1 API call instead of N
